src/bounds.rs

```diff
@@ -143,16 +143,103 @@ impl HexBounds {
 
 impl FromIterator<Hex> for HexBounds {
     fn from_iter<T: IntoIterator<Item = Hex>>(iter: T) -> Self {
-        let mut min = Hex::new(i32::MAX, i32::MAX);
-        let mut max = Hex::new(i32::MIN, i32::MIN);
-
-        for hex in iter {
-            min.x = min.x.min(hex.x);
-            max.x = max.x.max(hex.x);
-            min.y = min.y.min(hex.y);
-            max.y = max.y.max(hex.y);
+        let mut minx = i32::MAX;
+        let mut miny = i32::MAX;
+        let mut minz = i32::MAX;
+        let mut maxx = i32::MIN;
+        let mut maxy = i32::MIN;
+        let mut maxz = i32::MIN;
+
+        let count = iter
+            .into_iter()
+            .map(|hex| {
+                minx = minx.min(hex.x);
+                miny = miny.min(hex.y);
+                minz = minz.min(hex.z());
+                maxx = maxx.max(hex.x);
+                maxy = maxy.max(hex.y);
+                maxz = maxz.max(hex.z());
```

Comment:
```suggestion
        let mut min = IVec3::MAX;
        let mut max = IVec3::MIN;
       
        let count = iter
            .into_iter()
            .map(|hex| {
                min = min.min(hex.as_ivec3());
                max = max.max(hex.as_ivec3());
```
Nipick, we could gain in clarity using glam's `IVec3` struct and methods directly, for the same result.
If you prefer avoiding using `glam` we could use a [i32; 3] for `min` and `max`

Reply:
The reason why I used separate variables here is that it helped me think through the problem. These items should be *thought* about as separate entities. I think using a Glam struct should be fine for that as long as it is made clear what the algorithm is actually doing.

I'm test running swapping in the glam struct right now so we can compare.